rest_framework_gis/schema.py

```diff
@@ -0,0 +1,168 @@
+import warnings
+
+from django.contrib.gis.db import models
+from rest_framework.schemas.openapi import AutoSchema
+from rest_framework.utils import model_meta
+
+from rest_framework_gis.fields import GeometrySerializerMethodField
+from rest_framework_gis.serializers import GeoFeatureModelListSerializer, GeoFeatureModelSerializer
+
+
+class GeoFeatureAutoSchema(AutoSchema):
```

Comment:
Requiring users to set REST_FRAMEWORK.DEFAULT_SCHEMA_CLASS to a custom GIS specific class will limit the utility of this.  The default DRF schema generator is rarely used for serious projects because there are so many unsolved problems in it.  Most medium to large projects will use [drf-yasg](https://github.com/axnsan12/drf-yasg), or similar (including drf-spectacular which is new but quite good), and they use their own mechanism, either requiring that DEFAULT_SCHEMA_CLASS is set to their own class, or they ignore DEFAULT_SCHEMA_CLASS altogether.  And many projects need to sub-class DRF's AutoSchema to add their own behaviour.  I've tried to do a WIP for drf-spectacular to use the standard AutoSchema inheritance at https://github.com/tfranzel/drf-spectacular/pull/46 , but it has has some curly problems.

Keeping GeoFeatureAutoSchema is ok, but it would be better as a thin wrapper only, using functionality that exists outside of the class.

i.e. the schemas and functions that are embedded here inside this class should be in the module namespace, so they can be used with other generators.  The code in this custom AutoSchema only uses `self.` to reference the new code below -- it doesnt rely on any of the inner workings of DRF AutoSchema.

Reply:
1. Can you tell me what are the unresolved problems with the current master branch of DRF? https://github.com/encode/django-rest-framework/issues/7088 has listed 5 problems. 3 problems are already solved. Also, the DRF schema generator is maturing with time.
2. [drf-yasg](https://github.com/axnsan12/drf-yasg) is based on open API specification 2.0. Also, I would love to know which are the additional features that are present in drf-yasg but missing in the DRF schema generator.
3. Django-rest-framework-gis should override DEFAULT_SCHEMA_CLASS is the solution of Carlton Gibson, maintainer of schema generation part. Thread link: https://github.com/encode/django-rest-framework/issues/7202#issuecomment-591396557
4. All methods are starting with `_`.  I think it is a clear instruction to others, not to use these methods. Also, this is an unreviewed code. I have already requested @auvipy to review this code.

Reply:
I have no comment on DRF master.  I expected released versions of DRF to be usable, and in this case it isnt.  Usually the problem is that it falls apart when an installable app using DRF isnt 'perfect', which is 80%+ of the time, and often they are slowly maintained, so it will be a long time before they will work - usually it is exceptions all over the place.

While drf-yasg is openapi 2.0, it is usable for most people, and they dont care much about v2 vs v3

> Django-rest-framework-gis should override DEFAULT_SCHEMA_CLASS is the solution of Carlton Gibson

I am not saying that drf-gis shouldnt provide a AutoSchema class. I just hope that the core functionality of this PR is outside of the AutoSchema class, and the AutoSchema class is a thin wrapper. Especially the schema definitions, which are first class entities that drf-gis should expose.

Reply:
I have noted down your points. @auvipy Eagerly waiting for your review.

Reply:
I'm actually waiting for drf 3.12 release and celery 4.4.3[due by 5th of may]

Reply:
Thank you @auvipy for the quick reply. 
I have added a lot of extra code to make it compatible with DRF 3.10 and 3.11.
Should I remove that code and make schema generation only compatible with DRF 3.12?

Reply:
DRF 3.10 can be removed without any hesitation!